app/models/manageiq/providers/openshift/container_manager.rb

```diff
@@ -22,7 +24,26 @@ def self.event_monitor_class
 
   def create_project(project)
     connect.create_project_request(project)
-  rescue KubeException => e
```

Comment:
> Automation sometimes needs to catch specific exception, for example, it may need to get a status of an object. If the object does not exist in the provider, likely the API throws a provider specific object_not_found exception. Such exception should not fail the whole automation.

@bzwei this is not the case here, we're not doing anything smart with the exception (e.g. object does not exist in the provider => translate the exception in something useful for automation).

> When a method is created in automation we try to make it re-useable across multiple providers. That's the reason we recommend to wrap provider specific exceptions to a miq exception. If we start to wrap some exceptions, it's better to wrap all exceptions even though they may not be required.

@bzwei  if indeed you want to wrap **all** exception with no logic whatsoever then delegate this to the provider is not reliable and efficient: why do we have copy/paste these lines all over the place in all the providers:

```ruby
  ...
   rescue ... => e
    MiqException::MiqProvisionError, "Unexpected Exception while creating project: #{e}"
  end
```

Just do the wrapping in one place on the automation engine side.

Also, if from automation you can call any of the model methods, would you expect to write all the model methods so that they translate the exception to a `MiqException`?